monix-reactive/shared/src/main/scala/monix/reactive/internal/builders/CombineLatestListObservable.scala

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.reactive.internal.builders
+
+import monix.execution.Ack.{Continue, Stop}
+import monix.execution.cancelables.CompositeCancelable
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.Observable
+import monix.reactive.observers.Subscriber
+
+import scala.collection.mutable
+import scala.concurrent.Future
+import scala.util.Success
+import scala.util.control.NonFatal
+
+/** Only used in [[Observable.combineLatestList()]]. Is tested by `CombineLatestListSuite`. */
+private[reactive] final class CombineLatestListObservable[A, +R](observables: Seq[Observable[A]])(f: Seq[A] => R)
```

Comment:
I think we can get away without the `f` here.

`combineLatestMapX` helps avoid tuple allocations and induced primitive boxing, but we probably aren't buying much over `combineLatestList(...).map` as you do in your tests.

If you go with dropping it, you won't need to guard against user errors in f, so:
```scala
def rawOnNext(as: Seq[A]): Future[Ack] = if (isDone) Stop else out.onNext(as)
```